src/weather.cpp

```diff
@@ -66,6 +67,122 @@ static bool is_player_outside()
     return g->m.is_outside( point( g->u.posx(), g->u.posy() ) ) && g->get_levz() >= 0;
 }
 
+const weather_datum *weather::data( weather_type const type )
```

Comment:
As this function never returns a null pointer, it would be better to make it return a reference. Currently a future caller may do a null pointer check as they assume the returned pointer could be null.